src/index.ts

```diff
@@ -209,6 +219,60 @@ export function buildStorageKeySuffix(apiKey: string): string {
   return apiKey.replace(/\W/g, '').substring(0, 8);
 }
 
+/**
+ * Initializes the Eppo client with configuration parameters.
+ *
+ * The purpose is for use-cases where the configuration is available from an external process
+ * that can bootstrap the SDK.
+ *
+ * This method should be called once on application startup.
+ *
+ * @param config - client configuration
+ * @returns a singleton client instance
+ * @public
+ */
+export function initSync(config: IClientConfigSync): IEppoClient {
+  try {
+    const memoryOnlyConfigurationStore = configurationStorageFactory({
+      forceMemoryOnly: true,
+    });
+    memoryOnlyConfigurationStore.setEntries(config.flagsConfiguration);
+    EppoJSClient.instance.setConfigurationStore(memoryOnlyConfigurationStore);
+
+    // Allow the caller to override the default obfuscated mode, which is false
+    // since the purpose of this method is to bootstrap the SDK from an external source,
+    // which is likely a server that has not-obfuscated flag values.
+    EppoJSClient.instance.setIsObfuscated(config.isObfuscated ?? false);
+
+    if (config.assignmentLogger) {
+      EppoJSClient.instance.setLogger(config.assignmentLogger);
+    }
+
+    // There is no SDK key in the offline context.
+    const storageKeySuffix = buildStorageKeySuffix('no_api_key');
+
+    // As this is a synchronous initialization,
+    // we are unable to call the async `init` method on the assignment cache
+    // which loads the assignment cache from the browser's storage.
+    // Therefore there is no purpose trying to use a persistent assignment cache.
+    const assignmentCache = assignmentCacheFactory({
+      storageKeySuffix,
+      forceMemoryOnly: true,
+    });
+    EppoJSClient.instance.useCustomAssignmentCache(assignmentCache);
```

Comment:
There is still value to have an in-memory cache as it will remove duplicates for the duration of the session. Perhaps we can think creatively how to support this.